### PR TITLE
feat: doctolib.de germany compatibility

### PR DESCRIPTION
**Pull request by hschaeidt**
I am currently working on the full compatibility of both sites https://doctolib.fr/ and https://doctolib.de/ at the same time.

However, it seems that booking flows significantly differentiate between both platforms, as the health systems may vary from country to country.

Features:
- [x] Support most of the booking flows of both platforms
- [x] Extend the search terms to support both languages
- [x] Change the search terms to be inclusive instead of exclusive for better search result confidence
- [x] Remove the necessity of waiting for booking flow elements, by predefining precisely the possible flows in the code -> faster code execution until the date selection

Bugfixes:
- [x] Fix the date calculation and make it compatible with both regions
- [x] Get rid of the general public motive, as the vaccine is now open to the general public for a while

ToDo:
- [ ] Add translations for the german language, maybe we can build that on top of @nanls efforts in https://github.com/dunglas/vaccin.click/pull/33
- [x] Extract the rest of the workflow code into independent functions, to be more flexible on integrating more workflows if needed
- [ ] Think about, if this whole thing is somehow testable :)

### CODE REVIEW

File: content_scripts/doctolib/book.js [hunk 1]
```diff
@@ -344,22 +405,32 @@
...
         );
       }
+
+      const currentMonth = new Date().getMonth();
+      const selectedMonth = monthToNumber(parts[2]);
+      const selectedDay = parseInt(parts[1]);
+      const selectedYear = estimateYear(currentMonth, selectedMonth);
+      const selectedTime = parts[3];
+
       const date = new Date(
```
Comment (hschaeidt) [inline content_scripts/doctolib/book.js:432]: The date here is not really used for anything anymore, but this should fix it, there were little issues in the regex and the usage of `parts`. Additionally, the part with the year could have been incorrect when being at the end of a year, like now. The `estimateYear` function adds a year to the current one if the selected month is smaller than the current. This should work as long as we assume, that taking appointments in the past or in over a year should not be possible.

If I understand the code directly it was used before the 31st May 2020 to only search for available shots tomorrow. Should we remove the whole check?

Comment (dunglas) [reply content_scripts/doctolib/book.js:432]: Sure! This rule doesn't exist in France anymore.

**Review by dunglas — APPROVED** [review]

File: content_scripts/doctolib/book.js (hunk 1 above)
Comment (hschaeidt) [reply content_scripts/doctolib/book.js:433]: I'll leave it in for now, because it seems useful for https://github.com/dunglas/vaccin.click/pull/86 :)

**Comment by hschaeidt** [comment]
This PR is ready now. I rebased and tested it again. Unfortunately, I didn't have enough time to work on it for the last 2 weeks. For the 2 open Todos, I'd like to work on them in separate PRs. What do you think?

**Review by dunglas — APPROVED** [review]

**Comment by dunglas** [comment]
Thanks!

**Pull request merged**